standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -633,7 +634,8 @@ public Partition alterPartition(final RawStore msdb, Warehouse wh, final String
               }
 
               //rename the data directory
-              wh.renameDir(srcPath, destPath, true);
+              wh.renameDir(srcPath, destPath,
+                      ReplChangeManager.isSourceOfReplication(msdb.getDatabase(catName, dbname)));
```

Comment:
msdb.getDatabase can be stored earlier as it is used somewhere as well and shall get rid if NoSuchObjectException handling too.